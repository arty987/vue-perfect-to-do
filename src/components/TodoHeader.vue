<script>
export default {
  name: 'TodoHeader',
  props: {
    placeholder: {
      type: String,
      default: 'Title...',
    },
  },
  data() {
    return {
      inputValue: '',
    };
  },
  methods: {
    addTask() {
      if (this.inputValue.trim()) {
        this.$store.dispatch('addTask', this.inputValue);
        this.inputValue = '';
      }
    },
    clearInputValue() {
      this.inputValue = '';
    }
  }
}
</script>

<template>
  <li>
    <input
        type="checkbox"
        disabled
    />
    <input
        type="text"
        :placeholder="placeholder"
        v-model="inputValue"
        @keyup.enter="addTask"
        @keyup.esc="clearInputValue"
    />
  </li>
</template>

<style scoped>

</style>