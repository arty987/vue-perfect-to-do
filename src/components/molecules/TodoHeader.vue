<script>
import AppCheckbox from "@/components/atoms/AppCheckbox.vue";
import AppInput from "@/components/atoms/AppInput.vue";

export default {
  name: "TodoHeader",
  components: {
    AppInput,
    AppCheckbox
  },
}
</script>

<template>
  <li>
    <AppCheckbox :disabled="true"/>
    <AppInput/>
  </li>
</template>

<style scoped>

</style>