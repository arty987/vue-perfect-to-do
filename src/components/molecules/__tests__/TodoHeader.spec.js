import {shallowMount} from '@vue/test-utils'
import TodoHeader from '@/components/molecules/TodoHeader.vue';

describe('TodoHeader.vue', () => {
    it('Should render TodoHeader', () => {
        const wrapper = shallowMount(TodoHeader);
        expect(wrapper.element).toMatchSnapshot();
    });
})