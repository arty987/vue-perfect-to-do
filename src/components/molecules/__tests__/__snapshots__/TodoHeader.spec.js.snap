// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoHeader.vue Should render TodoHeader 1`] = `
<div
  class="d-flex justify-content-between align-items-center mb-4"
>
  <h1
    class="heading"
  >
    T O D O
  </h1>
  <img
    alt="logo"
    class="logo"
    src=""
  />
</div>
`;
